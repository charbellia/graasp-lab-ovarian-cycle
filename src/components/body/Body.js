import React from 'react';
import { connect } from 'react-redux';
import HumanBody from './human-body.png';
import { appendSvg } from '../../actions';

export class Body extends React.Component {
  componentDidMount() {
    const { dispatchAppendSvg } = this.props;
    dispatchAppendSvg(mapStateToProps);
  }

  render() {
    return (
      <div className="body-container">
        <img src={HumanBody} className="human-image" alt="Human Body" />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  svg: state.svg,
});

const mapDispatchToProps = {
  dispatchAppendSvg: appendSvg,
};

export default connect(mapStateToProps, mapDispatchToProps)(Body);
