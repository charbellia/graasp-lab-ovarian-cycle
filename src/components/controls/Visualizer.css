.Visualizer-Container {
  padding-top: 100px;
}
