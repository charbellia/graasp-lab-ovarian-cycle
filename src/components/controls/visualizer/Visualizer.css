.visualizer-container {
  background-color: #ffffff;
  border-left: 1px solid rgba(0, 0, 0, 0.12);
  height: 100%;
}
.fa-redo-alt {
  width: 2rem;
}
.observe-button, .test-button {
  width: 50%;
  padding-top: 1rem;
  padding-bottom: 1rem;
  border-radius: 0;
  font-size: 1.6rem;
  color: #000;
}
.test-button, .test-button:hover {
  border: 1px solid #dedede;
  color: #dedede;
  background-color: rgba(0,0,0,0);
}
.test-button-active, .observe-button-active {
  background-color: #dedede;
  font-weight: bold;
}
.test-button-active:hover,
.observe-button-active:hover {
  background-color: #EEEEEE;
  color: #555555;
}
#disabledBtn {
  cursor: help;
}
.btn, .modal-container button {
  outline: 0;
}
.modal-color-title, .modal-lang-title {
  font-size: 1.1rem;
}
.modal-lang-title {
  line-height: 2;
}
.modal-lang-container {
  margin-top: 1rem;
}
.modal-container .btn-secondary {
  position: fixed;
  bottom: 8%;
  right: 2%;
  background: #fafafa;
  border-color: #fafafa;
  border-radius: 50%;
  font-size: 1.8rem;
}
.modal-container .btn-secondary:active {
  background-color: #fafafa !important;
  border-color: #fafafa !important;
}
.modal-container .btn-secondary svg {
  margin-bottom: .3rem;
}
