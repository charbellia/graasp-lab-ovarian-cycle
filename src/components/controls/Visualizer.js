import React from 'react';
import Counter from './counter/Counter';
import Hormones from './hormones/Hormones';
import Phases from './phases/Phases';
import './Visualizer.css';

export const Visualizer = ({
  dayCount,
  handleStart,
  handleStop,
  ovulation,
  postOvulation,
  preOvulation,
  secretLhFsh,
  secretOestro,
  secretProgest,
}) => (
  <div className="Visualizer-Container">
    <Phases />
    <Counter
      handleStart={handleStart}
      handleStop={handleStop}
      dayCount={dayCount}
    />
    <Hormones
      ovulation={ovulation}
      postOvulation={postOvulation}
      preOvulation={preOvulation}
      secretLhFsh={secretLhFsh}
      secretProgest={secretProgest}
      secretOestro={secretOestro}
    />
  </div>
);

export default (Visualizer);
