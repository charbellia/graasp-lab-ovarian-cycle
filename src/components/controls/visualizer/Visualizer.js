import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { ToastContainer } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import Counter from '../counter/Counter';
import Phases from '../phases/Phases';
import Refresher from './refresher/Refresher';
import { AppState } from '../../../config/AppState';
import './Visualizer.css';
import {
  defaultLang,
  themeColor,
  titleState,
} from '../../../actions';
// importing all different child components and
// pass them params they need
export class Visualizer extends Component {
  state = AppState;

  handleChangeComplete = (color) => {
    const newColor = color.hex;
    const {
      dispatchThemeColor,
    } = this.props;
    dispatchThemeColor({ newColor });
    this.postMessage({
      themeColor: newColor,
    });
  }

  handleLang = (lang) => {
    const newLang = lang.value;
    const {
      dispatchDefaultLanguage,
    } = this.props;
    dispatchDefaultLanguage({ newLang });
    this.postMessage({
      defaultLang: newLang,
    });
  }

  onOpenModal = () => {
    this.setState({
      openModal: true,
    });
    this.postMessage({
      open_setting_modal: true,
    });
  }

  onCloseModal = () => {
    this.setState({
      openModal: false,
    });
    this.postMessage({
      open_setting_modal: false,
    });
  }

  postMessage = (data) => {
    const message = JSON.stringify(data);
    console.log('message', message);
    if (document.postMessage) {
      document.postMessage(message, '*');
    } else if (window.postMessage) {
      window.postMessage(message, '*');
    } else {
      console.error('unable to find postMessage');
    }
  };

  toggleTitle = () => {
    const { showTitle } = this.state;
    this.setState({ showTitle: !showTitle });
    const { dispatchTitleState } = this.props;
    dispatchTitleState({ showTitle });
    this.postMessage({
      show_title: showTitle,
    });
  }

  render() {
    const { openModal } = this.state;
    const {
      handleOvulation,
      handlePostOvulation,
      handlePreOvulation,
      reloadPage,
      handleStart,
      handleStop,
      isStarted,
      ovulationActive,
      postOvulationActive,
      t,
    } = this.props;
    return (
      <div className="visualizer-container">
        <Counter
          handleStart={handleStart}
          handleStop={handleStop}
          isStarted={isStarted}
          ovulationActive={ovulationActive}
          postOvulationActive={postOvulationActive}
          themeColor={themeColor}
          t={t}
        />
        <Phases
          handleOvulation={handleOvulation}
          handlePostOvulation={handlePostOvulation}
          handlePreOvulation={handlePreOvulation}
          t={t}
        />
        <ToastContainer autoClose={20000} />
        <Refresher
          reloadPage={reloadPage}
          onOpenModal={this.onOpenModal}
          onCloseModal={this.onCloseModal}
          openModal={openModal}
          handleChangeComplete={this.handleChangeComplete}
          handleLang={this.handleLang}
          toggleTitle={this.toggleTitle}
          t={t}
        />
      </div>
    );
  }
}

Visualizer.propTypes = {
  handleStart: PropTypes.func.isRequired,
  handleStop: PropTypes.func.isRequired,
  handleOvulation: PropTypes.func.isRequired,
  handlePostOvulation: PropTypes.func.isRequired,
  handlePreOvulation: PropTypes.func.isRequired,
  dispatchThemeColor: PropTypes.func.isRequired,
  dispatchDefaultLanguage: PropTypes.func.isRequired,
  dispatchTitleState: PropTypes.func.isRequired,
  t: PropTypes.func.isRequired,
  reloadPage: PropTypes.func.isRequired,
  isStarted: PropTypes.bool.isRequired,
  ovulationActive: PropTypes.bool.isRequired,
  postOvulationActive: PropTypes.bool.isRequired,
};

const mapStateToProps = state => ({
  themeColor: state.setting.themeColor,
});

const mapDispatchToProps = {
  dispatchThemeColor: themeColor,
  dispatchDefaultLanguage: defaultLang,
  dispatchTitleState: titleState,
};

const connectedComponent = connect(mapStateToProps, mapDispatchToProps)(Visualizer);

export default (connectedComponent);
