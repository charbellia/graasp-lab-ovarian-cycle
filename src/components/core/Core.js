import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Col, Row } from 'reactstrap';
import Body from '../body/Body';
import Hormones from '../controls/hormones/Hormones';
import Calendar from '../controls/calendar/Calendar';
import Visualizer from '../controls/visualizer/Visualizer';
import './Core.css';
// this is our sub main component
// it renders the body and all the part containing
// the dynamic informations of our application
export const Core = ({
  dayCount,
  isStarted,
  ovulation,
  ovulationActive,
  preOvulationActive,
  postOvulationActive,
  reloadPage,
  handleOvulation,
  handlePostOvulation,
  handlePreOvulation,
  handleStart,
  handleStop,
  postOvulation,
  preOvulation,
  secreteLhFsh,
  secreteOestro,
  secreteProgest,
  obserViewActive,
  themeColor,
  t,
}) => {
  const defaultColor = themeColor || '#0f94f8';
  return (
    <div className="core-container">
      <Row>
        <Col md={12}>
          <h1 className="lab-title" style={{ backgroundColor: defaultColor }}>{t('Synchronization of the ovarian cycle')}</h1>
        </Col>
      </Row>
      <Row>
        <Col sm="8">
          <Hormones
            ovulation={ovulation}
            postOvulation={postOvulation}
            preOvulation={preOvulation}
            secreteLhFsh={secreteLhFsh}
            secreteProgest={secreteProgest}
            secreteOestro={secreteOestro}
            t={t}
          />
          <Body />
        </Col>
        <Col sm="4">
          <Calendar
            dayCount={dayCount}
            themeColor={themeColor}
          />
          <Visualizer
            dayCount={dayCount}
            isStarted={isStarted}
            handleOvulation={handleOvulation}
            handlePostOvulation={handlePostOvulation}
            handlePreOvulation={handlePreOvulation}
            reloadPage={reloadPage}
            handleStart={handleStart}
            handleStop={handleStop}
            ovulation={ovulation}
            ovulationActive={ovulationActive}
            preOvulationActive={preOvulationActive}
            postOvulationActive={postOvulationActive}
            postOvulation={postOvulation}
            preOvulation={preOvulation}
            secreteLhFsh={secreteLhFsh}
            secreteOestro={secreteOestro}
            secreteProgest={secreteProgest}
            obserViewActive={obserViewActive}
            themeColor={themeColor}
            t={t}
          />
        </Col>
      </Row>
    </div>
  );
};

Core.propTypes = {
  dayCount: PropTypes.number.isRequired,
  handleStart: PropTypes.func.isRequired,
  handleStop: PropTypes.func.isRequired,
  handleOvulation: PropTypes.func.isRequired,
  handlePostOvulation: PropTypes.func.isRequired,
  handlePreOvulation: PropTypes.func.isRequired,
  t: PropTypes.func.isRequired,
  reloadPage: PropTypes.func.isRequired,
  isStarted: PropTypes.bool.isRequired,
  ovulation: PropTypes.bool.isRequired,
  ovulationActive: PropTypes.bool.isRequired,
  postOvulationActive: PropTypes.bool.isRequired,
  preOvulationActive: PropTypes.bool.isRequired,
  postOvulation: PropTypes.bool.isRequired,
  preOvulation: PropTypes.bool.isRequired,
  secreteLhFsh: PropTypes.bool.isRequired,
  secreteOestro: PropTypes.bool.isRequired,
  secreteProgest: PropTypes.bool.isRequired,
  obserViewActive: PropTypes.bool.isRequired,
  themeColor: PropTypes.string.isRequired,
};


const mapStateToProps = state => ({
  themeColor: state.setting.themeColor,
});

export default connect(mapStateToProps)(Core);
