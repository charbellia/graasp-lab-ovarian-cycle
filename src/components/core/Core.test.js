import React from 'react';
import { shallow } from 'enzyme';
import Core from './Core';
import Body from '../body/Body';
import Visualizer from '../controls/visualizer/Visualizer';

describe('<Core /> component', () => {
  const props = {
    dayCount: 0,
    handleStart: jest.fn(),
    handleStop: jest.fn(),
    handleOvulation: jest.fn(),
    handlePostOvulation: jest.fn(),
    handlePreOvulation: jest.fn(),
    reloadPage: jest.fn(),
    isStarted: false,
    ovulation: false,
    postOvulation: false,
    preOvulation: false,
    ovulationActive: false,
    postOvulationActive: false,
    preOvulationActive: false,
    secreteLhFsh: false,
    secreteOestro: false,
    secreteProgest: false,
    t: jest.fn(),
  };

  const component = shallow(<Core {...props} />);

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });
  it('has one <div /> with class core-container', () => {
    expect(component.find('div.core-container').length).toBe(1);
  });
  it('shows one <Body /> component', () => {
    expect(component.find(Body).length).toEqual(1);
  });
  it('has one <Visualizer /> component', () => {
    expect(component.find(Visualizer).length).toBe(1);
  });
});
