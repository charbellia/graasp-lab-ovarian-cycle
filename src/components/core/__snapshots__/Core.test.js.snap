// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Core /> component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Core
    dayCount={0}
    handleOvulation={[MockFunction]}
    handlePostOvulation={[MockFunction]}
    handlePreOvulation={[MockFunction]}
    handleStart={[MockFunction]}
    handleStop={[MockFunction]}
    isStarted={false}
    ovulation={false}
    ovulationActive={false}
    postOvulation={false}
    postOvulationActive={false}
    preOvulation={false}
    preOvulationActive={false}
    reloadPage={[MockFunction]}
    secreteLhFsh={false}
    secreteOestro={false}
    secreteProgest={false}
    t={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Row
        tag="div"
      >
        <Col
          sm="8"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Connect(Body) />
        </Col>
        <Col
          sm="4"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Visualizer
            dayCount={0}
            handleOvulation={[MockFunction]}
            handlePostOvulation={[MockFunction]}
            handlePreOvulation={[MockFunction]}
            handleStart={[MockFunction]}
            handleStop={[MockFunction]}
            isStarted={false}
            ovulation={false}
            ovulationActive={false}
            postOvulation={false}
            postOvulationActive={false}
            preOvulation={false}
            preOvulationActive={false}
            reloadPage={[MockFunction]}
            secreteLhFsh={false}
            secreteOestro={false}
            secreteProgest={false}
            t={[MockFunction]}
          />
        </Col>
      </Row>,
      "className": "core-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Col
            sm="8"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Connect(Body) />
          </Col>,
          <Col
            sm="4"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Visualizer
              dayCount={0}
              handleOvulation={[MockFunction]}
              handlePostOvulation={[MockFunction]}
              handlePreOvulation={[MockFunction]}
              handleStart={[MockFunction]}
              handleStop={[MockFunction]}
              isStarted={false}
              ovulation={false}
              ovulationActive={false}
              postOvulation={false}
              postOvulationActive={false}
              preOvulation={false}
              preOvulationActive={false}
              reloadPage={[MockFunction]}
              secreteLhFsh={false}
              secreteOestro={false}
              secreteProgest={false}
              t={[MockFunction]}
            />
          </Col>,
        ],
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Connect(Body) />,
            "sm": "8",
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Visualizer
              dayCount={0}
              handleOvulation={[MockFunction]}
              handlePostOvulation={[MockFunction]}
              handlePreOvulation={[MockFunction]}
              handleStart={[MockFunction]}
              handleStop={[MockFunction]}
              isStarted={false}
              ovulation={false}
              ovulationActive={false}
              postOvulation={false}
              postOvulationActive={false}
              preOvulation={false}
              preOvulationActive={false}
              reloadPage={[MockFunction]}
              secreteLhFsh={false}
              secreteOestro={false}
              secreteProgest={false}
              t={[MockFunction]}
            />,
            "sm": "4",
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "dayCount": 0,
              "handleOvulation": [MockFunction],
              "handlePostOvulation": [MockFunction],
              "handlePreOvulation": [MockFunction],
              "handleStart": [MockFunction],
              "handleStop": [MockFunction],
              "isStarted": false,
              "ovulation": false,
              "ovulationActive": false,
              "postOvulation": false,
              "postOvulationActive": false,
              "preOvulation": false,
              "preOvulationActive": false,
              "reloadPage": [MockFunction],
              "secreteLhFsh": false,
              "secreteOestro": false,
              "secreteProgest": false,
              "t": [MockFunction],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Row
          tag="div"
        >
          <Col
            sm="8"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Connect(Body) />
          </Col>
          <Col
            sm="4"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Visualizer
              dayCount={0}
              handleOvulation={[MockFunction]}
              handlePostOvulation={[MockFunction]}
              handlePreOvulation={[MockFunction]}
              handleStart={[MockFunction]}
              handleStop={[MockFunction]}
              isStarted={false}
              ovulation={false}
              ovulationActive={false}
              postOvulation={false}
              postOvulationActive={false}
              preOvulation={false}
              preOvulationActive={false}
              reloadPage={[MockFunction]}
              secreteLhFsh={false}
              secreteOestro={false}
              secreteProgest={false}
              t={[MockFunction]}
            />
          </Col>
        </Row>,
        "className": "core-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Col
              sm="8"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <Connect(Body) />
            </Col>,
            <Col
              sm="4"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <Visualizer
                dayCount={0}
                handleOvulation={[MockFunction]}
                handlePostOvulation={[MockFunction]}
                handlePreOvulation={[MockFunction]}
                handleStart={[MockFunction]}
                handleStop={[MockFunction]}
                isStarted={false}
                ovulation={false}
                ovulationActive={false}
                postOvulation={false}
                postOvulationActive={false}
                preOvulation={false}
                preOvulationActive={false}
                reloadPage={[MockFunction]}
                secreteLhFsh={false}
                secreteOestro={false}
                secreteProgest={false}
                t={[MockFunction]}
              />
            </Col>,
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Connect(Body) />,
              "sm": "8",
              "tag": "div",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Visualizer
                dayCount={0}
                handleOvulation={[MockFunction]}
                handlePostOvulation={[MockFunction]}
                handlePreOvulation={[MockFunction]}
                handleStart={[MockFunction]}
                handleStop={[MockFunction]}
                isStarted={false}
                ovulation={false}
                ovulationActive={false}
                postOvulation={false}
                postOvulationActive={false}
                preOvulation={false}
                preOvulationActive={false}
                reloadPage={[MockFunction]}
                secreteLhFsh={false}
                secreteOestro={false}
                secreteProgest={false}
                t={[MockFunction]}
              />,
              "sm": "4",
              "tag": "div",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "dayCount": 0,
                "handleOvulation": [MockFunction],
                "handlePostOvulation": [MockFunction],
                "handlePreOvulation": [MockFunction],
                "handleStart": [MockFunction],
                "handleStop": [MockFunction],
                "isStarted": false,
                "ovulation": false,
                "ovulationActive": false,
                "postOvulation": false,
                "postOvulationActive": false,
                "preOvulation": false,
                "preOvulationActive": false,
                "reloadPage": [MockFunction],
                "secreteLhFsh": false,
                "secreteOestro": false,
                "secreteProgest": false,
                "t": [MockFunction],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Core /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Core
    dayCount={0}
    handleOvulation={[MockFunction]}
    handlePostOvulation={[MockFunction]}
    handlePreOvulation={[MockFunction]}
    handleStart={[MockFunction]}
    handleStop={[MockFunction]}
    isStarted={false}
    ovulation={false}
    ovulationActive={false}
    postOvulation={false}
    postOvulationActive={false}
    preOvulation={false}
    preOvulationActive={false}
    reloadPage={[MockFunction]}
    secreteLhFsh={false}
    secreteOestro={false}
    secreteProgest={false}
    t={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Row
        tag="div"
      >
        <Col
          sm="8"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Connect(Body) />
        </Col>
        <Col
          sm="4"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Visualizer
            dayCount={0}
            handleOvulation={[MockFunction]}
            handlePostOvulation={[MockFunction]}
            handlePreOvulation={[MockFunction]}
            handleStart={[MockFunction]}
            handleStop={[MockFunction]}
            isStarted={false}
            ovulation={false}
            ovulationActive={false}
            postOvulation={false}
            postOvulationActive={false}
            preOvulation={false}
            preOvulationActive={false}
            reloadPage={[MockFunction]}
            secreteLhFsh={false}
            secreteOestro={false}
            secreteProgest={false}
            t={[MockFunction]}
          />
        </Col>
      </Row>,
      "className": "core-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Col
            sm="8"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Connect(Body) />
          </Col>,
          <Col
            sm="4"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Visualizer
              dayCount={0}
              handleOvulation={[MockFunction]}
              handlePostOvulation={[MockFunction]}
              handlePreOvulation={[MockFunction]}
              handleStart={[MockFunction]}
              handleStop={[MockFunction]}
              isStarted={false}
              ovulation={false}
              ovulationActive={false}
              postOvulation={false}
              postOvulationActive={false}
              preOvulation={false}
              preOvulationActive={false}
              reloadPage={[MockFunction]}
              secreteLhFsh={false}
              secreteOestro={false}
              secreteProgest={false}
              t={[MockFunction]}
            />
          </Col>,
        ],
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Connect(Body) />,
            "sm": "8",
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Visualizer
              dayCount={0}
              handleOvulation={[MockFunction]}
              handlePostOvulation={[MockFunction]}
              handlePreOvulation={[MockFunction]}
              handleStart={[MockFunction]}
              handleStop={[MockFunction]}
              isStarted={false}
              ovulation={false}
              ovulationActive={false}
              postOvulation={false}
              postOvulationActive={false}
              preOvulation={false}
              preOvulationActive={false}
              reloadPage={[MockFunction]}
              secreteLhFsh={false}
              secreteOestro={false}
              secreteProgest={false}
              t={[MockFunction]}
            />,
            "sm": "4",
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "dayCount": 0,
              "handleOvulation": [MockFunction],
              "handlePostOvulation": [MockFunction],
              "handlePreOvulation": [MockFunction],
              "handleStart": [MockFunction],
              "handleStop": [MockFunction],
              "isStarted": false,
              "ovulation": false,
              "ovulationActive": false,
              "postOvulation": false,
              "postOvulationActive": false,
              "preOvulation": false,
              "preOvulationActive": false,
              "reloadPage": [MockFunction],
              "secreteLhFsh": false,
              "secreteOestro": false,
              "secreteProgest": false,
              "t": [MockFunction],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Row
          tag="div"
        >
          <Col
            sm="8"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Connect(Body) />
          </Col>
          <Col
            sm="4"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Visualizer
              dayCount={0}
              handleOvulation={[MockFunction]}
              handlePostOvulation={[MockFunction]}
              handlePreOvulation={[MockFunction]}
              handleStart={[MockFunction]}
              handleStop={[MockFunction]}
              isStarted={false}
              ovulation={false}
              ovulationActive={false}
              postOvulation={false}
              postOvulationActive={false}
              preOvulation={false}
              preOvulationActive={false}
              reloadPage={[MockFunction]}
              secreteLhFsh={false}
              secreteOestro={false}
              secreteProgest={false}
              t={[MockFunction]}
            />
          </Col>
        </Row>,
        "className": "core-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Col
              sm="8"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <Connect(Body) />
            </Col>,
            <Col
              sm="4"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <Visualizer
                dayCount={0}
                handleOvulation={[MockFunction]}
                handlePostOvulation={[MockFunction]}
                handlePreOvulation={[MockFunction]}
                handleStart={[MockFunction]}
                handleStop={[MockFunction]}
                isStarted={false}
                ovulation={false}
                ovulationActive={false}
                postOvulation={false}
                postOvulationActive={false}
                preOvulation={false}
                preOvulationActive={false}
                reloadPage={[MockFunction]}
                secreteLhFsh={false}
                secreteOestro={false}
                secreteProgest={false}
                t={[MockFunction]}
              />
            </Col>,
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Connect(Body) />,
              "sm": "8",
              "tag": "div",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Visualizer
                dayCount={0}
                handleOvulation={[MockFunction]}
                handlePostOvulation={[MockFunction]}
                handlePreOvulation={[MockFunction]}
                handleStart={[MockFunction]}
                handleStop={[MockFunction]}
                isStarted={false}
                ovulation={false}
                ovulationActive={false}
                postOvulation={false}
                postOvulationActive={false}
                preOvulation={false}
                preOvulationActive={false}
                reloadPage={[MockFunction]}
                secreteLhFsh={false}
                secreteOestro={false}
                secreteProgest={false}
                t={[MockFunction]}
              />,
              "sm": "4",
              "tag": "div",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "dayCount": 0,
                "handleOvulation": [MockFunction],
                "handlePostOvulation": [MockFunction],
                "handlePreOvulation": [MockFunction],
                "handleStart": [MockFunction],
                "handleStop": [MockFunction],
                "isStarted": false,
                "ovulation": false,
                "ovulationActive": false,
                "postOvulation": false,
                "postOvulationActive": false,
                "preOvulation": false,
                "preOvulationActive": false,
                "reloadPage": [MockFunction],
                "secreteLhFsh": false,
                "secreteOestro": false,
                "secreteProgest": false,
                "t": [MockFunction],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
